### PR TITLE
Add a fee address to stable swap

### PR DESCRIPTION
**Pull request by baboobhaiya**

### CODE REVIEW

File: contracts/stableswap/Swap.sol [hunk 1]
```diff
@@ -41,6 +41,9 @@ contract Swap is OwnerPausableUpgradeable, ReentrancyGuardUpgradeable {
     // getTokenIndex function also relies on this mapping to retrieve token index.
     mapping(address => uint8) private tokenIndexes;
 
+    // decides the fee address
+    address public feeAddress;
```
Comment (baboobhaiya) [inline contracts/stableswap/Swap.sol:45]: Check if the variable is set correctly

File: contracts/stableswap/Swap.sol [hunk 2]
```diff
@@ -88,6 +91,8 @@ contract Swap is OwnerPausableUpgradeable, ReentrancyGuardUpgradeable {
         uint256 futureTime
     );
     event StopRampA(uint256 currentA, uint256 time);
+    event FeeAddressChanged(address newFeeAddress, address oldFeeAddress);
+
```
Comment (baboobhaiya) [inline contracts/stableswap/Swap.sol:95]: Check if the event is emitted

File: contracts/stableswap/Swap.sol [hunk 3]
```diff
@@ -179,6 +184,9 @@ contract Swap is OwnerPausableUpgradeable, ReentrancyGuardUpgradeable {
         // swapStorage.futureATime = 0;
         swapStorage.swapFee = _fee;
         swapStorage.adminFee = _adminFee;
+
+        // initialize feeAddress to be the owner
+        feeAddress = owner();
```
Comment (baboobhaiya) [inline contracts/stableswap/Swap.sol:189]: Check if owner is the feeAddress

File: contracts/stableswap/Swap.sol [hunk 4]
```diff
@@ -481,10 +497,13 @@ contract Swap is OwnerPausableUpgradeable, ReentrancyGuardUpgradeable {
     /*** ADMIN FUNCTIONS ***/
 
     /**
-     * @notice Withdraw all admin fees to the contract owner
+     * @notice Update the fee address.
+     * @param newFeeAddress new fee address to be applied for fee withdrawal
      */
-    function withdrawAdminFees() external onlyOwner {
-        swapStorage.withdrawAdminFees(owner());
+    function changeFeeAddress(address newFeeAddress) external onlyOwner {
```
Comment (baboobhaiya) [inline contracts/stableswap/Swap.sol:503]: Only owner can change this variable
Check if variable is changed
Check if the event is emitted

**Review by rej156 — APPROVED** [review]

**Pull request merged**